### PR TITLE
Macpie/close submit

### PR DESCRIPTION
**Pull request by macpie**

### CODE REVIEW

**Comment by macpie** [comment]
Fix for https://github.com/helium/router/issues/562

File: include/metrics.hrl [hunk 1]
```diff
@@ -47,6 +48,8 @@
     {?METRICS_VM_CPU, prometheus_gauge, [cpu], "Router CPU usage"},
     {?METRICS_VM_PROC_Q, prometheus_gauge, [name], "Router process queue"},
     {?METRICS_VM_ETS_MEMORY, prometheus_gauge, [name], "Router ets memory"},
-    {?METRICS_XOR_FILTER, prometheus_gauge, [], "Router XOR Filter udpates"},
+    {?METRICS_XOR_FILTER, prometheus_counter, [], "Router XOR Filter udpates"},
```
Comment (michaeldjeffrey) [inline include/metrics.hrl:51]: This is gauge because the metric is the cost in DC of an XOR txn.

Comment (macpie) [reply include/metrics.hrl:51]: Easier to read I think if a counter where it only increment with a new submit, no?

File: src/metrics/router_metrics.erl [hunk 2]
```diff
@@ -127,7 +128,12 @@ ws_state(State) ->
 
 -spec xor_filter_update(DC :: non_neg_integer()) -> ok.
 xor_filter_update(DC) ->
-    _ = prometheus_gauge:set(?METRICS_XOR_FILTER, DC),
+    _ = prometheus_counter:inc(?METRICS_XOR_FILTER, DC),
```
Comment (michaeldjeffrey) [inline src/metrics/router_metrics.erl:131]:
```suggestion
    _ = prometheus_counter:inc(?METRICS_XOR_FILTER, [], DC),
```

I increment by the amount of DC spent.

Comment (michaeldjeffrey) [reply src/metrics/router_metrics.erl:131]: This is already covered when the second argument is not a list.

**Review by michaeldjeffrey — APPROVED** [review]

**Pull request merged**